src/lib/DatePicker.svelte

```diff
@@ -7,19 +7,36 @@
 
   const dispatch = createEventDispatcher<{ select: undefined }>()
 
-  /** Date value */
-  export let value = new Date()
+  /** Date value. It's `null` if no date is selected */
+  export let value: Date | null = null
   function setValue(d: Date) {
-    if (d.getTime() !== value.getTime()) value = d
+    if (d.getTime() !== value?.getTime()) {
+      value = d
+    }
   }
   function updateValue(updater: (date: Date) => Date) {
-    let d = updater(new Date(value.getTime()))
-    setValue(d)
+    const newValue = updater(new Date(shownDate.getTime()))
+    setValue(newValue)
   }
+
+  /** Default Date to use */
+  const defaultDate = new Date()
+
+  let shownDate = value ?? defaultDate
+  $: shownDate = value ?? defaultDate
```

Comment:
```suggestion
  /** The date shown in the popup, for when `value` is null */
  let shownDate = value ?? defaultDate
  $: if (value) shownDate = value
```

We shouldn't reset the `shownDate` to default when the value is set to `null`

Reply:
Fixed this on my `nullable` branch